continuous-integration/requirements.Rmd

```diff
@@ -0,0 +1,28 @@
+---
+title: "Requirements for continuous integration"
+maintainer: "Thierry Onkelinx"
+output: html_document
+---
+
+# Must have
+
+- start automatically when pushing a git repository on Github or Bitbucket
+- start automatically when creating a pull request on Github or Bitbucket
+- able to use a Docker image to setup the environment in which CI runs
+- able to set environment variables
+- able to check code style with lintr and report the results back to the SCM
+- able to check code coverage with covr and report the results back to the SCM
+- a succesful build creates a Docker image
+- a succesful build on specific branches can be pushed to Docker hub (e.g. https://hub.docker.com/r/inbonatura2000/n2khelper/tags/)
+- configuration of CI can be partially through a configuration file and partially through a webapplication
```

Comment:
I agree that a much a possible should be defined in configuration files contained within a repository. However, some information is to sensitive (e.g. usernames, passwords, tokens) to be placed in the configuration file. I'd rather add a file to the repository which describes how the CI was setup.